### PR TITLE
deps: update golang to 1.20.11

### PR DESCRIPTION
**Pull request by lysliu**
Bump golang from 1.20.10 to 1.20.11
PR check failed e.g. https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6808420299/job/18512839352?pr=1564
Vulnerability #1: GO-2023-2186
    Incorrect detection of reserved device names on Windows in path/filepath
  More info: https://pkg.go.dev/vuln/GO-2023-2186
  Standard library
    Found in: path/filepath@go1.20.10
    Fixed in: path/filepath@go1.21.4

Fixes: #1565

### CODE REVIEW

**Comment by huoqifeng** [comment]
I'm not sure whether we're ready to upgrade to 1.21 yet, I guess a more safe approach is to update to `go1.20.11`.

**Review by liudalibj — APPROVED** [review]
LGTM

**Review by liudalibj — CHANGES_REQUESTED** [review]
please use `hack/update-go-container.sh` to update the version.

File: hack/Dockerfile.golang [hunk 1]
```diff
@@ -5,7 +5,7 @@ FROM --platform=$TARGETPLATFORM ${BASE_IMAGE} as base
 
 # DO NOT UPDATE THIS BY HAND !!
 # Use hack/update-go-container.sh to update the version and hashes.
-ARG GO_VERSION=1.20.10
+ARG GO_VERSION=1.20.11
 ARG GO_LINUX_ARM64_SHA256=fb3c7e15fc4413c5b81eb9f26dbd7cd4faedd5c720b30fa8e2ff77457f74cab6
```
Comment (liudalibj) [inline hack/Dockerfile.golang:9]: these SHA256 hash need be updated too, please use `hack/update-go-container.sh`

**Comment by lysliu** [comment]
> I'm not sure whether we're ready to upgrade to 1.21 yet, I guess a more safe approach is to update to `go1.20.11`.

Thanks, actually `1.21.4` failed some testcase, and `go1.20.11` also fixed the Vulnerability https://github.com/confidential-containers/cloud-api-adaptor/pull/1: GO-2023-2186.

**Review by liudalibj — APPROVED** [review]
LGTM

**Review by genjuro214 — APPROVED** [review]
LGTM

**Pull request merged**